configure.ac

```diff
@@ -50,21 +50,6 @@ AS_IF([test x"$with_build_id" = x"no"], [with_build_id=""])
 AC_DEFINE_UNQUOTED([BUILD_ID],["$with_build_id"],
                    [adds build_id to version if it was defined])
 
-AC_ARG_ENABLE([debug],
-	      [AS_HELP_STRING([--enable-debug],
-			      [Enable debugging @<:@default=no@:>@])
-	      ],
-	      [],
-	      [enable_debug=no])
-
-AS_IF([test x"$enable_debug" != x"no"],
-      [dbg=1
-       CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} ${debug_c_other_flags} $CFLAGS"],
```

Comment:
@shefty @a-ilango @gladkovdmitry17 The three of you touched lines around this area, so I wanted to bring this to your attention.

This PR is because @a-ilango's a4e954c07d commit added `-fstack-protector-strong` into `$debug_c_other_flags`.  However, RHEL 6 and 7 default gcc do not recognize this flag.  So I moved this block down and added some logic to make it check to see if the flag is supported before adding it to CFLAGS.  This has to be done *after* the C compiler is determined so that we can use it to determine if the given flag(s) is(are) supported.

However, this may have a side effect: I'm not quite sure what the intent was for mixing `--enable-debug` with the default value for CFLAGS (in the block slightly below this, with the comment "Fix autoconf's habit of setting CFLAGS="-g -O2" by default while still allowing the user to explicitly set CFLAGS=""").

I.e., moving this block down may defeat what you guys were trying to do there -- I'm not sure...?

Reply:
I just received a complaint about fstack-protector-strong this morning.  :)

Reply:
Does moving this block down affect your intent with the default value of CFLAGS?

Reply:
I think we need to wait for Dmitry to respond for this question.

Reply:
> Does moving this block down affect your intent with the default value of CFLAGS?

Thanks. This work for us.